### PR TITLE
[flutter_tools] rename library to be less absurd

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Instead of targets/dart.dart -> targets/common.dart

### CODE REVIEW

File: packages/flutter_tools/lib/src/build_system/targets/android.dart [hunk 1]
```diff
@@ -207,7 +207,7 @@ class AndroidAot extends AotElfBase {
 
   @override
   List<Source> get inputs => <Source>[
-    const Source.pattern('{FLUTTER_ROOT}/packages/flutter_tools/lib/src/build_system/targets/dart.dart'),
+    const Source.pattern('{FLUTTER_ROOT}/packages/flutter_tools/lib/src/build_system/targets/android.dart'),
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/build_system/targets/android.dart:210]: Can we rely on a build stamp so it invalidates on every upgrade?  Build system behavior could change without this file changing (could be a different PR).

**Review by jmagman — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/build_system/targets/android.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_system/targets/android.dart:210]: The engine version is used as a shortcut for this: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/build_system/build_system.dart#L318

It should also include `flutter/version` though.

The goal of this input is just for local development, not to handle upgrades
